ovl_En_Jsjutan with 1 non_matching

AngheloAlf opened the pull request:
The non_matching function only has regalloc issues.

~I also tried to make an xml for the texture and vertices in data, but I can't export the whole data with the xml neither I can just `#include` only the `Vtx[]`, so the xml is half baked.~ Almost all of this was solved by last ZAPD update.

~Edit: There is currently one remaining issue in the xml, and this [PR](https://github.com/zeldaret/ZAPD/pull/92) can fix it, so don't merge this yet.~ This is fixed now.

petrie911 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+
+    this->dyna.actor.shape.rot.x = Math_SinS(globalCtx->gameplayFrames * 3000) * 300.0f;
+    this->dyna.actor.shape.rot.z = Math_CosS(globalCtx->gameplayFrames * 3500) * 300.0f;
+    if (!globalCtx) {}
```
This can probably be removed with globalCtx2

petrie911 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+            phi_s0->v.ob[1] = phi_s2->v.ob[1] = this->dyna.actor.floorHeight;
+            this->dyna.actor.world.pos = actorPos;
+        }
+        phi_s0++;
+        phi_s2++;
```
These can go in the for loop iterator.

petrie911 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+
+static s32 sUnused[2] = {
+    0,
+    0,
```
```suggestion
    0
```
we probably don't need the formatter to do this

petrie911 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+        phi_s0_3->n.n[0] = spFC.x;
+        phi_s0_3->n.n[1] = spFC.y;
+        phi_s0_3->n.n[2] = spFC.z;
+        phi_s0_3++;
```
this can go in for loop iterator

petrie911 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,455 @@ const ActorInit En_Jsjutan_InitVars = {
+    sp108.y = 0.0f;
+    sp108.z = 120.0f;
+
+    for (j = 0; j < 0x90; j++) {
```
All of these carpet-related numbers should be in decimal, as the carpet is 12 x 12.

AngheloAlf commented:
I made the changes you requested. Tell me if i missed something!

fig02 commented on assets/xml/overlays/ovl_En_Jsjutan.xml:
```diff
@@ -0,0 +1,38 @@
+            <Vtx/>
+        </Array>
+
+        <!--D_80A8E400--> <!-- Needs a fix in ZAPD. See https://github.com/zeldaret/ZAPD/pull/85 -->
+        <!--Collision Name="D_80A8E400" Offset="0x4C60"/-->
```
now that the fixes are in this should be good to go.
though, you mention in the pr comment that the xml is half baked. if thats the case id leave it out of the pr and include it when we can fix all of the remaining issues

AngheloAlf replied:
I just checked, and with the current ZAPD version the only problem with that xml is a dlist that references a variable declared in bss. That variable can't be found during the compilation process (because the autogenerated C file isn't #including any of the actor files), so the compiler barks.
I'll try to see what I can do in ZAPD to workaround it.

AngheloAlf replied:
I made a PR in ZAPD that solves this problem, so I think we should wait for it.

fig02 commented on spec:
```diff
@@ -2529,8 +2529,6 @@ endseg
     include "build/src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.o"
-    include "build/data/overlays/actors/z_en_jsjutan.data.o"
-    include "build/data/overlays/actors/z_en_jsjutan.bss.o"
     include "build/data/overlays/actors/z_en_jsjutan.reloc.o"
```
make sure to make a non_matching ifdef for reloc

fig02 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,462 @@ const ActorInit En_Jsjutan_InitVars = {
+        this->unk_170 = 1000.0f;
+    } else {
+        Math_ApproachF(&this->dyna.actor.world.pos.y, this->unk_168 - 1000.0f, 1.0f, this->dyna.actor.velocity.y);
+        if (!gSaveContext.dayTime) {
+            if (1) {}
```
daytime commonly needs the (void)0 trick. could help with the temp above and some of these other memes. something to keep in mind for the next time you try to match this at least

AngheloAlf replied:
Do I have to come back here?
![image](https://user-images.githubusercontent.com/7416381/111086532-4ee1a300-84fb-11eb-87de-1cedcea3d52e.png)

fig02 replied:
lmfao you can leave it as a note so that whoever tries this later will consider it

AngheloAlf commented:
This should be ready now.

I deleted the file `z_en_jsjutan_data.c` since it had a texture and vtx. It is extracted by ZAPD now. 
I also renamed the variables of the extracted data.

Roman971 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.h:
```diff
@@ -7,10 +7,20 @@
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x2C];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x0165 */ u8 unk_164; // flag to set carpet shadow textures.
```
```suggestion
    /* 0x0164 */ u8 unk_164; // flag to set carpet shadow textures.
```

Roman971 commented on src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:
```diff
@@ -21,15 +27,466 @@ const ActorInit En_Jsjutan_InitVars = {
+    s32 pad;
+    CollisionHeader* header = NULL;
+
+    this->dyna.actor.flags = this->dyna.actor.flags & ~1;
```
```suggestion
    this->dyna.actor.flags &= ~1;
```
